src/core/src/FabricMaterial.cpp

```diff
@@ -320,67 +326,58 @@ void FabricMaterial::setMaterial(int64_t tilesetId, const MaterialInfo& material
     auto srw = UsdUtil::getFabricStageReaderWriter();
 
     setShaderValues(_shaderPath, materialInfo);
-    setTilesetId(tilesetId);
+
+    for (const auto& path : _allPaths) {
+        FabricUtil::setTilesetId(path, tilesetId);
```

Comment:
Would it ever be the case that a a prim and its path belongs to multiple tilesets? For example the shader in `_allPaths.push_back(shaderPath)`

Reply:
It shouldn't ever happen because unique IDs are given to every FabricGeometry / FabricMaterial that's created.

* https://github.com/CesiumGS/cesium-omniverse/blob/9137f544f67989daf3de0f8d587b4b71cbdd520a/src/core/src/FabricResourceManager.cpp#L74
* https://github.com/CesiumGS/cesium-omniverse/blob/9137f544f67989daf3de0f8d587b4b71cbdd520a/src/core/src/FabricGeometryPool.cpp#L26

Reply:
And all the inner prims like shaders are appended to the material path.

Reply:
I think I see. "Creating" seems like what we'd call "cloning" in Unity.